### ISSUE TITLE
Bootstrap button contrast not compliant with WCAG (not accessible)

### ISSUE THREAD

**Issue by mandudeboy** [jhipster/generator-jhipster#7961]
##### **Overview of the issue**

Some of the buttons used throughout the GUI, for example for "view" and "edit" in the entity pages, fail the contrast requirements from WCAG 2.0 level AA.

##### **Motivation for or Use Case**

For anyone using jHipster in projects that have accessibility requirements, the background colours need to be manually modified to comply with regulations.

##### **Reproduce the error**

Create an entity of any kind using JDL and import it. 
View the entity page. 
Create an entity.
Verify that the "view" button in the table doesn't have good enough contrasts using a contrast checking tool like https://webaim.org/resources/contrastchecker/  

##### **Suggest a Fix**
Changing the background color in css for btn-info and btn-primary to a bit darker solves the problem. 
For "view" button (btn-info), #128091 would comply
For "edit" button (btn-primary), #317AB9 would comply

##### **JHipster Version(s)**

v4.14.3

##### **Browsers and Operating System**

All browsers.

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
Can you do aPR with your suggestions?

**Comment by JazzBrotha** [comment]
I'll grab this one if it's ok?

**Comment by jdubois** [comment]
Oh yes @JazzBrotha please do, thanks!

**Comment by jdubois** [comment]
@JazzBrotha I'm adding you to a project team so the ticket can be assigned to you

**Comment by pascalgrimaud** [comment]
done @JazzBrotha

**Comment by JazzBrotha** [comment]
I have discovered that there is also a color contrast issue with the generated bootstrap navbar, The link text color does not have sufficient contrast to the navbar background color. Do you want me to fix this in the same branch or do you want a separate GitHub issue for this?

**Comment by deepu105** [comment]
You can do in same branch

Thanks & Regards,
Deepu

**Comment by jdubois** [comment]
- We're going to drop CSS and only keep Sass in #8150 so I would mostly concentrate on Sass
- Why do we even re-define those colors? If we just remove them, we will have the default Bootstrap colors, which should be good (I guess they did their homework on this), and then our users can work from a "standard" Bootstrap configuration in order to make their own theme

**Comment by antonioortizpola** [comment]
I also think that the default bootstrap colors should be used, what was the reason in the first place to change them?, It is a little confusing when you start changing the layout (also the buttons changing size on hover are weird)

**Comment by deepu105** [comment]
Let me look at this later, originally I just updated primary colors to
match Jhipster theme which should be ok

**Comment by mandudeboy** [comment]
>  If we just remove them, we will have the default Bootstrap colors, which should be good (I guess they did their homework on this)

No, as is common in accessibility the default is not 100% compliant, and no - they did not do their homework (at least not all of it).

This issue is not the "fault" of jHipster, but it's up to us to fix it if we want jHipster to be compliant.

**Comment by deepu105** [comment]
While accessibility is very important I'm not willing to override everything provided by Bootstrap. 
See my reason below
1. Our policy is to use the defaults from the underlying lib as much as possible.
2. The default theme provided by JHipster is just a starter and we are not expecting any real application to use it as it is. So, in that case, it's up to the users to use colors based on there specific requirements and accessibility goals.
3. If you using the default theme as it is, its highly the case that its either some demo, personal project or some small internal app. Anything that has a wide audience will always have its own UX design, theme, and those apps always customize them.
4. We have never seen any real jHipster application going to prod with the default theme we provide.
5. We made Sass the default and it is extremely easy to customize colors with Sass. In the future we will be Sass only

So, in the end, the accessibility compliance on the default theme is not as important as it seems to be. Hence I'm closing this.
